### PR TITLE
4dgs updated events

### PR DESCRIPTION
**Pull request by Deb-chatt**
made the changes for the Loss function, maybe confirm if the rendered images are being generated properly for the event loss function. will probably need to change the colmap to json file[you mentioned to check the LLFF codebase for that].

### CODE REVIEW

**Comment by PiotrWolinski** [comment]
About the poses: [here](https://github.com/Fyusion/LLFF/blob/c6e27b1ee59cb18f054ccb0f87a90214dbe70482/llff/poses/pose_utils.py#L11) you can find how LLFF loads colmap data (so the format that our data follows as well). Then you can see that [here](https://github.com/PiotrWolinski/4DGaussians/blob/c099e73c96606425f676e89687850e15ea9e0560/multipleviewprogress.sh#L117) it is being used to generate `poses_bounds.npy` file which holds information about all the poses.

File: scripts/colmap_to_json.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+# Add project root to path to import colmap_loader
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
+from scene.colmap_loader import read_extrinsics_binary, read_intrinsics_binary, qvec2rotmat
+
+def c2w_from_colmap(R, t):
+    w2c = np.eye(4)
+    w2c[:3, :3] = R
+    w2c[:3, 3] = t
+    c2w = np.linalg.inv(w2c)
+    c2w[:3, 1:3] *= -1  # OpenCV to OpenGL coordinate system conversion
```
Comment (PiotrWolinski) [inline scripts/colmap_to_json.py:17]: So I think it would make sense to confirm that this method produces results consistent with LLFF, [this line](https://github.com/Fyusion/LLFF/blob/c6e27b1ee59cb18f054ccb0f87a90214dbe70482/llff/poses/pose_utils.py#L51) specifically.

File: scene/static_cam_dataset.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            if not os.path.exists(image_path):
+                print(f"Warning: File not found: {image_path}")
+                continue
+
+            
+            image_paths.append(image_path)
+            image_poses.append((R, T))
+            image_times.append(float(idx / len(self.transforms_data["frames"])))
+
+        if self.split == "train":
```
Comment (PiotrWolinski) [inline scene/static_cam_dataset.py]: I think it would make sense to test this approach with splitting the dataset against the one present in the multiview dataset. The one here is based on `llffhold` and assigns every `llffhold` image to the test set, so in the end it will have size `num_images // llffhold`. Default `llffhold` is 8, so for our short sequences it can be quite a lot test images (taken away from the training data). [Here](https://github.com/hustvl/4DGaussians/blob/843d5ac636c37e4b611242287754f3d4ed150144/scene/multipleview_dataset.py#L45) you can see that it just assumes 3 images per camera to be taken into test set.

**Review by PiotrWolinski — COMMENTED** [review]
Looks much better compared to the previous revision, I hope that it will work better too ✌🏼
